### ISSUE TITLE
Feature Request: Responsive content sizing in EzCards

### ISSUE THREAD

**Issue by pgrippi** [ezcater/recipe#544]
We are writing a responsive EzCard where we have a card image that has different sizes for desktop, tablet, and mobile. When EzCard is not given an `imageMaxWidth` it will use 50% of the card for the image and 50% for the content:

<img width="942" alt="Screen Shot 2021-03-19 at 5 33 44 PM" src="https://user-images.githubusercontent.com/1180841/111844363-46adad00-88d9-11eb-9a10-8a604a65ebd2.png">

When we give EzCard an `imageMaxWidth` it will always take up that amount of space, so we have a weird gap that looks like this:

<img width="946" alt="Screen Shot 2021-03-19 at 5 26 54 PM" src="https://user-images.githubusercontent.com/1180841/111843994-8d4ed780-88d8-11eb-912b-3774d392cf07.png">
<img width="849" alt="Screen Shot 2021-03-19 at 5 28 09 PM" src="https://user-images.githubusercontent.com/1180841/111844000-8f189b00-88d8-11eb-83c4-ab776d4758a6.png">

Could we allow for `<EzContent>` to fill the remaining card space when `imagePosition` is left or right and it's paired with an `<EzPreview>` component?

**Comment by CraigCav** [comment]
I *think* what is being asked for here is for `EzCard` to support a responsive value for `imageMaxWidth` instead of a fixed value.

@pgrippi could you confirm whether this is the case?

**Comment by CraigCav** [comment]
I've put together an [example](https://ezcater.github.io/recipe/playroom/?code=N4Igxg9gJgpiBcIA8BRAXgBQIYHMYD4AdAOwAJTU0BhLAJylIEsBbXGDCAZ0YBdGJiAXmDAARlk4x4pAOQ8IABxkAaUsxhRGAV2bSZAGxgAzHjIC%2BZpqzwBZLAA8A6oyg8AFoMIgATAFYApF5EZOQU6Bi0MABujDAA7sGhoUgKjGA8WpGJScmcEJlgMGoajFieIAAUrPYAtHEu7tIA7ACcLQr2AJRepJy0YADKMDzC9e4Rxoz2FTIA9CxsnLM8WJCcAHQAVgo4Mp2Ws9k5FHkFReqaZV5VjMR1DW7SABwADC8d3SC9-UMjwGNuCZGKYzebWGBLFZrGpuRg4NxbHZ7A5HHJIFg4EjHJJ9MCjXiAyLA6ZzBZ4SGrLiI3b7LHY8hYfQjLxDQzpfhkCBGUgAISwmywpAAYoxOG5SFCuKQsMQGDyYMZeXRxJBBQAVSmcLx0%2BmcHgAT0MwmAOvppHqrkesgAjG9-CpTfS3DA4W4eHpbS97cpHdiIKJNjB0iL3bJIFEYLQHSF6RZfeRDr6kLNUulMgRTcnwpEYvFUZQABIwLCwWio8iUAAyWH1%2BR4pH0Nbr5U4Cn0vCC8bCaCLJduOF6jDQMHKAGYepwtKJnX3iDhygBVBScMCMooAQS07hgxD4YGFjaiEFonZj2PXzC0hnkpBsMHsaRlpA1okMXazPeLmjn5eS6B5W7yGQWiSOUPCRnwdD6qeZonAoT6wlAsDEPgd7EFoyatjKv5JEgnBRAOuZxDyED2OULykBRo4AGykDRPTOq6zI%2BC8PQWu45TeKxXx0KUMIuMhYG0FoMA9EYEBgCBWCviOXhGIykgwbBFBgIw-SGKQYD6uU1HrNaPRgGRXjWk86zeD0tBjusTxBMmqnqRmZ70kg9lgBpWnlNaTTWQZRkgCZZkWVZNkgPgdlqW5jnKSpEXudpXjeC06y%2BL5nmmeZXyWV4o4%2BaF4UOThySzPhOCFR%2BAE8EBv4ftWtZbvmszoL2pb5ugVACOBu7VQohWOEUCi0C4RT5LQkj6BGnCkAIvSRjEc4Ss6ahcPWWBbs6u5pLe96PmQ4nQFNW7cLAU3cneD6QOsvJbuaMAyJEXZxHQxD9qQdXSAIhSvfkEq0PqB20KQe1QKodUGPopDENEkakKIRTSRpN7HqWgO0BAzDSsQEDbgDZ07V2kR6qttAyjwl0AJKmJN7hYPWOAQNAACEpBk7KV54EwnKZKQUS8cMf1chKmpTQDI2A4ex5GFepBuPkki9ApWCXX1XarsQcjmlgvCA8eX1aBKEA-X9vAM0mKYNW1HU7jwrVoEK9PgWWSboLVdYNk2W7lIN8I8EpxyUBVQGkCBskgJIkCylBQQ0MQhT6OVgECGV-4J8BoFeANCy-VHAjArQzDx5Viem877vW5mjW2-bkbZB%2BND0MEH7YHgwQgGYQA) demonstrating what i'd expect this to look like with Recipe today.

Note that `imageMaxWidth` does not apply when `imagePosition="top"`, which allows for the correct sizing for images on 'left|right" positioned images, and still maintain full-width images for `imagePosition="top"`.

Please let us know if there is anything else you need over-and-above what we provide today.

**Comment by pgrippi** [comment]
> I _think_ what is being asked for here is for `EzCard` to support a responsive value for `imageMaxWidth` instead of a fixed value.
> 
> @pgrippi could you confirm whether this is the case?

yes, I believe that was the intention - to be able to specify different widths (in pixels) for different viewports

> I've put together an example demonstrating what i'd expect this to look like with Recipe today.

I'm not sure this meets our needs. In your example you used a percentage width, on search we use a specific width in pixels - which is where this breaks down. We use the same image for all three views. So something like this:

```
<EzCard imagePosition={{base: 'top', medium: 'left'}} imageMaxWidth={{medium: '150px', large: '200px'}}>
```

**Comment by CraigCav** [comment]
There are a couple of things that make this tricky.

Internally, Recipe uses CSS variables to vary the width of the CSS grid area. Of course IE11 doesn't support CSS vars, so we rely on the capabilities of the CSS vars ponyfill; which only supports `:root` level variables, and not the local overrides necessary to easily implement this feature.

This subsequently means that in order to vary the width of the grid in IE, we need to dynamically insert CSS for `-ms-grid-columns` with interpolated values for the column width. 

Making this responsive would mean that we also have to dynamically generate media query selectors also with interpolated values for the column width at each breakpoint.

This is definitely possible, but significantly more work than the alternative. 

As an aside, dynamically generating CSS rules is typically more expensive than inserting static values, since these styles are less likely to result in a cache-hit. Dynamically generated rules also present a hurdle to future static CSS extraction.
